server/src/main/scala/za/co/absa/atum/server/api/repository/BaseRepository.scala

```diff
@@ -17,35 +17,57 @@
 package za.co.absa.atum.server.api.repository
 
 import za.co.absa.atum.server.api.exception.DatabaseError
-import za.co.absa.fadb.exceptions.StatusException
+import za.co.absa.db.fadb.exceptions.StatusException
+import za.co.absa.db.fadb.status.{FailedOrRow, FailedOrRows}
 import zio._
 
 trait BaseRepository {
 
-  def dbCall[R](
-    dbFuncCall: Task[R],
-    operationName: String
-  ): IO[DatabaseError, R] = {
-    dbFuncCall
-      .zipLeft(ZIO.logDebug(s"Operation '$operationName' succeeded in database"))
-      .mapError(error => DatabaseError(error.getMessage))
-      .tapError(error => ZIO.logError(s"Operation '$operationName' failed: ${error.message}"))
-  }
-
-  def dbCallWithStatus[R](
-    dbFuncCall: Task[Either[StatusException, R]],
-    operationName: String
-  ): IO[DatabaseError, Either[StatusException, R]] = {
+  private def logOperationResult[R](operationName: String, dbFuncCall: Task[R]): ZIO[Any, Throwable, R] = {
     dbFuncCall
       .tap {
-        case Left(statusException) =>
+        case Left(statusException: StatusException) =>
           ZIO.logError(
             s"Exception caused by operation: '$operationName': " +
-              s"(${statusException.status.statusCode}) ${statusException.status.statusText}"
+              s"(${statusException.status.statusCode}), ${statusException.status.statusText}"
           )
         case Right(_) => ZIO.logDebug(s"Operation '$operationName' succeeded in database")
+        case _ => ZIO.logError(s"Operation '$operationName' did not return an Either")
```

Comment:
It does not make much sense to me. You should require the effects on its input to be Task[FailedOrRow[R]] or Task[FailedOrRows[R]] (another method).